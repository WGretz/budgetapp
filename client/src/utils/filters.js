const filters = {
  budgetToCurrency (value) {
    var formatter = new Intl.NumberFormat('en-US', {
      style: 'currency',
      currency: 'USD',
      minimumFractionDigits: 2
    });
    if (value === 0)
      return formatter.format(0)
    return formatter.format(-1*value/100)
  }
}

module.exports = filters

