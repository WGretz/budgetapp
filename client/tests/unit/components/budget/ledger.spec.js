import { shallowMount, createLocalVue } from '@vue/test-utils'
import Ledger from '@/components/Budget/ledger'
import { budgetToCurrency } from '@/utils/filters'

describe('Ledger`.vue', () => {
  let localVue
  beforeEach(() => {
    localVue = createLocalVue()
    localVue.filter('budgetToCurrency', budgetToCurrency)
  })
  it('renders props.msg when passed', () => {

    const wrapper = shallowMount(Ledger, { localVue })

    wrapper.setData({
      transactions: [
        { id: 1, description: "Betelgeuse's Bakery For The Recently deceased", amountInCents: 999 },
        { id: 2, description: "CBGBs", amountInCents: 4269 }
      ]
    })
    expect(wrapper.element).toMatchSnapshot(this)
  })
})
