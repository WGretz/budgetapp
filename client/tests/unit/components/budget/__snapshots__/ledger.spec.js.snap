// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ledger\`.vue renders props.msg when passed 1`] = `
<div>
  <table>
    <tr>
      <td>
        Betelgeuse's Bakery For The Recently deceased
      </td>
       
      <td>
        -$9.99
      </td>
    </tr>
    <tr>
      <td>
        CBGBs
      </td>
       
      <td>
        -$42.69
      </td>
    </tr>
  </table>
</div>
`;
